pilot/pkg/networking/core/v1alpha3/listener_builder.go

```diff
@@ -584,24 +590,44 @@ func newHTTPPassThroughFilterChain(configgen *ConfigGeneratorImpl,
 	return filterChains
 }
 
-func newTCPProxyOutboundListenerFilter(push *model.PushContext, node *model.Proxy) *listener.Filter {
+func buildFallthroughNetworkFilters(push *model.PushContext, node *model.Proxy) []*listener.Filter {
 	tcpProxy := &tcp_proxy.TcpProxy{
 		StatPrefix:       util.BlackHoleCluster,
 		ClusterSpecifier: &tcp_proxy.TcpProxy_Cluster{Cluster: util.BlackHoleCluster},
 	}
+	requireSniForwarding := false
 	if util.IsAllowAnyOutbound(node) {
 		// We need a passthrough filter to fill in the filter stack for orig_dst listener
+		egressCluster := util.PassthroughCluster
+
+		// no need to check for nil value as the previous if check has checked
+		if node.SidecarScope.OutboundTrafficPolicy.EgressProxy != nil {
+			// user has provided an explicit destination for all the unknown traffic.
+			// build a cluster out of this destination
+			egressCluster = istio_route.GetDestinationCluster(node.SidecarScope.OutboundTrafficPolicy.EgressProxy,
```

Comment:
Is EgressProxy.port guaranteed to be set?

Reply:
There are two options in my opinion:
1. `EgressProxy.Port` *MUST* be set. Enforce the constraint by making changes in the validation logic for `Destination`. It may break cases where service has only one port and it works without setting Port in Destination.
2. Best effort approach -- look for service by matching EgressProxy.Host . If service found, pass service to GetDestinationCluster otherwise pass nil and wish for success